### PR TITLE
Netlify (including Autofill)

### PR DESCRIPTION
**Pull request by afjlambert**

### CODE REVIEW

File: apps/client/src/__mocks__/context.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default ({
   const context = useContext(Context);
   context.topic = topics.find((t) => t.slug === topicMock);
   if (addressMock) {
-    context.address = addressMock;
+    context.autofillData.address = addressMock;
```
Comment (robinpiets) [inline apps/client/src/__mocks__/context.js]: IMHO I prefer `address` over `autofillData` because address is way more clear what it is and autofillData requires background info (for new users and reviewers). Can we not rename to `userData` or something more general?

File: apps/client/src/autofill.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { getRestrictionByTypeName } from "./utils";
+
+export const autofillResolvers = {
+  monumentBoolean: ({ address }) =>
+    address?.restrictions
+      ? !!getRestrictionByTypeName(address.restrictions, "Monument")
+      : undefined,
```
Comment (robinpiets) [inline apps/client/src/autofill.js]: Can this not be `address?.restrictions && !!getRestrictionByTypeName(address.restrictions, "Monument")` or do you need to specifically return `undefined`?

File: apps/client/src/autofill.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      );
+      if (monumentData) {
+        return `"${monumentData.name}"`;
+      }
+    }
+  },
+  cityScape: ({ address }) =>
+    address?.restrictions
+      ? !!getRestrictionByTypeName(address.restrictions, "CityScape")
+      : undefined,
```
Comment (robinpiets) [inline apps/client/src/autofill.js]: Same as above ^^

File: apps/client/src/autofill.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+export const autofillMap = {
+  monumentList: "address",
+  monumentBoolean: "address",
+  cityScape: "address",
+  // geo: 'map', // for trees ?
```
Comment (robinpiets) [inline apps/client/src/autofill.js]: Remove empty comment for future feature

File: apps/client/src/components/AddressData.js [hunk 5]
```diff
@@ -1,58 +1,60 @@
...
 
   return (
-    <div>
-      <AddressResult title="Monument:">
-        <Paragraph>
-          {monument ? `Ja. ${monument}` : "Nee. Geen monument"}
-        </Paragraph>
-      </AddressResult>
+    <StyledAddressData>
+      <Paragraph strong style={{ marginBottom: "0px" }}>
```
Comment (robinpiets) [inline apps/client/src/components/AddressData.js]: `gutterBottom={0}` not working? Otherwise `{{ marginBottom: 0 }}` or `{{ marginTop: 20 }}` also works without making it a string

File: apps/client/src/hoc/withTopic.js [hunk 6]
```diff
@@ -15,16 +18,21 @@ const withTopic = (Component) => () => {
...
+      return <RedirectPage topic={topic} />;
+    }
+
     context.topic = topic;
-    return <Component topic={topic} />;
+    return <Component {...props} topic={topic} />;
   }
-  return <NotFoundPage />;
+
+  return <NotFoundPage {...props} />;
```
Comment (robinpiets) [inline apps/client/src/hoc/withTopic.js]: NotFoundPage doesnt need props

File: apps/client/src/hoc/withTopic.test.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+// };
+
+const { MemoryRouter } = ReactRouter;
+
+// jest.doMock("react-router-dom", () => ({
+//   ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
+//   useParams: () => {
+//     return { slug: "with-redirect" };
+//   }
+// }));
```
Comment (robinpiets) [inline apps/client/src/hoc/withTopic.test.js]: WIP?

File: apps/client/src/pages/IntroPage.js [hunk 8]
```diff
@@ -19,11 +20,11 @@ const IntroPage = ({ topic: { text, slug, intro } }) => {
       <Suspense fallback={<Loading />}>
         <Intro />
       </Suspense>
-      <Form action={geturl(routes.location, { slug })}>
+      <Form action={geturl(routes.questions, topic)}>
```
Comment (robinpiets) [inline apps/client/src/pages/IntroPage.js]: Why is this a redirect to `routes.questions`?

File: apps/client/src/pages/LocationPage.js [hunk 9]
```diff
@@ -40,8 +40,9 @@ const LocationPage = ({ topic }) => {
         action: `postcode - ${slug.replace("-", " ")}`,
         name: address.postalCode.substring(0, 4),
       });
-      context.address = address;
-      history.push(geturl(routes.address, { slug }));
+
+      context.autofillData.address = address;
```
Comment (robinpiets) [inline apps/client/src/pages/LocationPage.js]: We only store `address` in `autofillData`, why not keep it `address`? Is this for future features?

In the refresh branch this is changed to `context.address[slug]` because we're saving different address per slug. Don't know if this helps thinking about different naming. Let's discuss?

File: apps/client/src/pages/LocationPage.js [hunk 10]
```diff
@@ -40,8 +40,9 @@ const LocationPage = ({ topic }) => {
...
+      history.push(geturl(routes.address, topic));
```
Comment (robinpiets) [inline apps/client/src/pages/LocationPage.js]: Why is this then pushing to `routes.address`?

File: apps/client/src/pages/QuestionsPage.js [hunk 11]
```diff
@@ -73,8 +69,20 @@ const QuestionsPage = ({ topic, checker }) => {
...
+    }
+  };
+
+  const goBack = () => {
+    // Go back to Location page if needed or intropage otherwise
+    const dataNeed = checker.getAutofillDataNeeds(autofillMap).shift();
+
+    // See if any questions have data needs
+    if (dataNeed) {
+      history.push(geturl(autofillRoutes[dataNeed], { slug }));
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.js]: Here comes the bug with going back from Question Page to Address Page...

File: apps/client/src/pages/ResultsPage.js [hunk 12]
```diff
@@ -38,6 +41,12 @@ const ResultsPage = ({ topic, checker }) => {
         }}
         data-testid={RESULTS_PAGE}
       >
+        <Paragraph>
+          Vergunningcheck uitgevoerd voor: <AddressLine address={address} />
+        </Paragraph>
+
+        <AddressData displayZoningPlans={false} address={address} />
```
Comment (robinpiets) [inline apps/client/src/pages/ResultsPage.js]: `displayZoningPlans={false}` is default so this can be removed

File: apps/client/src/sttr_client/models/checker.js [hunk 13]
```diff
@@ -118,14 +106,87 @@ class Checker {
     return this.rewindTo(this._stack.length - (this._done === true ? 1 : 2));
   }
 
+  /**
+   * XXX
```
Comment (robinpiets) [inline apps/client/src/sttr_client/models/checker.js]: WIP?

File: apps/graphql/app.js [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+const config = require("./config");
+const { server } = require("./src/graphql");
+const secure = require("./src/secure");
+const development = require("./src/development");
+
+const app = secure(express());
+if (process.env.NODE_ENV !== "production") {
+  development(app);
+}
+
```
Comment (robinpiets) [inline apps/graphql/app.js:14]: Maybe it was a better idea to separate the Autofill and Netlify PR's. Now the whole context changes.

**Review by robinpiets — APPROVED** [review]
Some minor improvements. Overall very nice. Including some great refactors!

It should have been nicer to separate the Netlify and Autofill because the PR is massive and the Netlify part of this PR is hard to understand

**Comment by robinpiets** [comment]
Also, this PR is now merging with develop branch, is that the plan? What about the `feature/refresh` branch?

File: apps/client/src/__mocks__/context.js (hunk 1 above)
Comment (svenjens) [reply apps/client/src/__mocks__/context.js]: why not put it straight in the context? 
and just get const { autofillAddress } = useContext(Context); ?
 
And i thought we are getting the address for each activity?

File: apps/client/src/components/AddressDataStyles.js [hunk 15]
```diff
@@ -1,7 +1,7 @@
 import styled from "styled-components";
 import { themeColor } from "@datapunt/asc-ui";
 
-export const StyledAddressResult = styled.div`
+export const StyledAddressData = styled.div`
```
Comment (svenjens) [inline apps/client/src/components/AddressDataStyles.js]: Why did you rename this? Imo data could be anywhere in the application

File: apps/client/src/hoc/withAutofillData.js [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+import withChecker from "./withChecker";
+import Context from "../context";
+import { routes, geturl, autofillRoutes } from "../routes";
+import { isEmptyObject } from "../utils";
+import { autofillResolvers, autofillMap } from "../autofill";
+
+const withAutofillData = (Component) =>
+  withChecker((props) => {
+    const history = useHistory();
+    const { autofillData } = useContext(Context);
```
Comment (svenjens) [inline apps/client/src/hoc/withAutofillData.js]: Not for now, but lets be consistent, with the destructs from useContext. 
I perfer this way above const context = useContext(Context)

File: apps/client/src/hoc/withTopic.test.js [hunk 17]
```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { render } from "../utils/test-utils";
+import ReactRouter from "react-router-dom";
+import withTopic from "./withTopic";
+
+// ReactRouter.useParams = () => {
+//   console.log("doe het nu!");
```
Comment (svenjens) [inline apps/client/src/hoc/withTopic.test.js]: Doet het niet denk ik

File: apps/client/src/pages/LocationPage.js [hunk 18]
```diff
@@ -67,9 +68,9 @@ const LocationPage = ({ topic }) => {
         <LocationFinder
           setAddress={setAddress}
           setErrorMessage={setErrorMessage}
-          postalCode={context.address?.postalCode}
-          houseNumberFull={context.address?.houseNumberFull}
-          houseNumber={context.address?.houseNumberFull}
+          postalCode={context.autofillData.address?.postalCode}
```
Comment (svenjens) [inline apps/client/src/pages/LocationPage.js]: Maybe destructure this?

Comment (robinpiets) [reply apps/client/src/pages/LocationPage.js]: Don't waste time on this destructure, the whole thing is going to be changed with session data instead of context data:
```
postalCode={sessionAddress?.postalCode}
houseNumberFull={sessionAddress?.houseNumberFull}
houseNumber={sessionAddress?.houseNumberFull}
```

File: apps/client/src/__mocks__/context.js (hunk 1 above)
Comment (afjlambert) [reply apps/client/src/__mocks__/context.js]: Discussed this in today's dev-meeting. We'll refactor the context-variables later.

File: apps/client/src/autofill.js (hunk 2 above)
Comment (afjlambert) [reply apps/client/src/autofill.js]: Good catch, will fix.

File: apps/client/src/autofill.js (hunk 3 above)
Comment (afjlambert) [reply apps/client/src/autofill.js]: will do.

File: apps/client/src/components/AddressData.js (hunk 5 above)
Comment (afjlambert) [reply apps/client/src/components/AddressData.js]: will fix.

File: apps/client/src/hoc/withTopic.js (hunk 6 above)
Comment (afjlambert) [reply apps/client/src/hoc/withTopic.js]: check.

File: apps/client/src/autofill.js (hunk 4 above)
Comment (afjlambert) [reply apps/client/src/autofill.js]: thanks

File: apps/client/src/pages/ResultsPage.js (hunk 12 above)
Comment (afjlambert) [reply apps/client/src/pages/ResultsPage.js]: true.

File: apps/graphql/app.js (hunk 14 above)
Comment (afjlambert) [reply apps/graphql/app.js:14]: you are correct. We'll do this differently in the future.

File: apps/client/src/components/AddressDataStyles.js (hunk 15 above)
Comment (afjlambert) [reply apps/client/src/components/AddressDataStyles.js]: Will fix.

File: apps/client/src/hoc/withAutofillData.js (hunk 16 above)
Comment (afjlambert) [reply apps/client/src/hoc/withAutofillData.js]: Please explain why in dev-meeting. I'm curious.

File: apps/client/src/hoc/withTopic.test.js (hunk 17 above)
Comment (afjlambert) [reply apps/client/src/hoc/withTopic.test.js]: Haha lol. Will remove this.

File: apps/client/src/sttr_client/models/checker.js (hunk 13 above)
Comment (afjlambert) [reply apps/client/src/sttr_client/models/checker.js]: will fix.

File: apps/client/src/hoc/withTopic.test.js (hunk 7 above)
Comment (afjlambert) [reply apps/client/src/hoc/withTopic.test.js]: check
